controllers/extendeddaemonset/controller.go

```diff
@@ -361,7 +361,7 @@ func (r *Reconciler) selectNodes(logger logr.Logger, daemonset *datadoghqv1alpha
 		currentNodes = canaryStatus.Nodes
 	}
 
-	nbCanaryPod, err := intstrutil.GetValueFromIntOrPercent(daemonsetSpec.Strategy.Canary.Replicas, int(replicaset.Status.Desired), true)
+	nbCanaryPod, err := intstrutil.GetScaledValueFromIntOrPercent(daemonsetSpec.Strategy.Canary.Replicas, int(daemonset.Status.Desired), true)
```

Comment:
Using EDS desired replicas is necessary here, as the canary ERS desired field is 0 at this point (so if using a percentage, we'd get `nbCanaryPod = 0` always)